appengine/pubsub/app.rb

```diff
@@ -55,6 +58,27 @@
 end
 # [END gae_flex_pubsub_push]
 
+# [START gaeflex_net_pubsub_auth_push]
+post "/pubsub/authenticated-push" do
+  halt 400 if params[:token] != PUBSUB_VERIFICATION_TOKEN
+
+  bearer = request.env["HTTP_AUTHORIZATION"]
+  token = bearer.split(" ")[1]
+  begin
+    claim = Google::Auth::IDTokens.verify_oidc token, aud: "example.com"
+    claims.push claim
```

Comment:
Why are we accumulating the claims into a big array? (Note: this could grow memory without bound if the instance remains active for long enough.) I see this in the other language samples as well, but I couldn't find anything in the cloudsite page that discusses what we're doing with these. If we're not using this, I'd drop it from the sample. (If we emphatically want to keep it, then we should move the initialization of the claims variable above, into a region tag that is visible on cloudsite so people understand what it is.)

Reply:
What you said is fair! Perhaps I can provide some background. 

The main purpose of these samples is to show how to verify a JWT from Pub/Sub, and what it looks like (thus the accumulating and the printing). 

Default push messages from Pub/Sub doesn't have a JWT attached, when authenticated push is enabled, we attach a JWT, and users can verify it to make sure the request is from Pub/Sub.

Reply:
@anguillanneuf I copied this from the other samples, is this line a required part of the sample?

Reply:
Are you asking about `claims.push claim`?

Reply:
It is not strictly!

Reply:
I added printing a UL of claims to the index view.

Reply:
@anguillanneuf It looks like the Python index view is missing `ul` elements for bearer tokens and claims: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/appengine/standard_python3/pubsub/templates/index.html#L24-L31

Reply:
Good catch! let me quickly add `ul` elements. Thanks Chris.